packages/typescript-estree/src/convert.ts

```diff
@@ -2583,6 +2583,24 @@ export class Converter {
 
       case SyntaxKind.InterfaceDeclaration: {
         const interfaceHeritageClauses = node.heritageClauses ?? [];
+        const interfaceExtends: TSESTree.TSInterfaceHeritage[] = [];
+
+        for (const heritageClause of interfaceHeritageClauses) {
+          if (heritageClause.token !== SyntaxKind.ExtendsKeyword) {
+            throw createError(
+              this.ast,
+              heritageClause.pos,
+              "Interface declaration cannot have 'implements' clause.",
+            );
+          }
+
+          interfaceExtends.push(
+            ...heritageClause.types.map(
+              n => this.convertChild(n, node) as TSESTree.TSInterfaceHeritage,
+            ),
+          );
```

Comment:
This just avoids the `...` spread arg which is good when you don't know exactly how many arguments there will be (it's possible to stackoverflow otherwise!)
Also very minor perf improvement because it avoids the arrow function closure cost.

```suggestion
          for (const heritageType of heritageClause.types) {
            interfaceExtends.push(
              this.convertChild(heritageType, node) as TSESTree.TSInterfaceHeritage,
            );
          }
```